model/device.go

```diff
@@ -431,3 +432,13 @@ func DeviceIsUp(ctx context.Context, store datastore.Store, mac string) (bool, e
 	}
 	return false, nil
 }
+
+// IsValidStatus returns true if the supplied number is a valid status, or false otherwise.
+func IsValidStatus(n int64) bool {
+	switch n {
+	case DeviceStatusOK, DeviceStatusUpdate, DeviceStatusReboot, DeviceStatusDebug, DeviceStatusUpgrade, DeviceStatusAlarm, DeviceStatusTest, DeviceStatusShutdown:
```

Comment:
Optional, but there's a pattern I've seen for iota based enums where you add an "end" value to the enum so that you can do this,
```
func IsValidStatus(n int64) bool {
    if 0 <= n && n < deviceStatusEnd {
        return true
    }
    return false
}
```

Then when you add any more values you don't have to update the validation function.